sections/main-product.liquid

```diff
@@ -444,6 +546,107 @@
   }
 
   customElements.define('product-modal', ProductModal);
+
+  class MediaGallery extends HTMLElement {
+    constructor() {
+      super();
+      this.elements = {
+        liveRegion: this.querySelector('[id^="GalleryStatus"]'),
+        viewer: this.querySelector('[id^="GalleryViewer"]'),
+        thumbnails: this.querySelector('[id^="GalleryThumbnails"]')
+      }
+      this.mql = window.matchMedia('(min-width: 750px)');
+      if (!this.elements.thumbnails) return;
+
+      this.elements.viewer.addEventListener('slideChanged', debounce(this.onSlideChanged.bind(this), 500));
+      this.elements.thumbnails.querySelectorAll('[data-target]').forEach((mediaToSwitch) => {
+        mediaToSwitch.querySelector('button').addEventListener('click', this.setActiveMedia.bind(this, mediaToSwitch.dataset.target, false));
+      });
+      if (this.dataset.desktopLayout !== 'stacked' && this.mql.matches) this.removeListSemantic();
+    }
+
+    onSlideChanged(event) {
+      const thumbnail = this.elements.thumbnails.querySelector(`[data-target="${ event.detail.currentElement.dataset.mediaId }"]`);
+      this.setActiveThumbnail(thumbnail);
+    }
+
+    setActiveMedia(mediaId, prepend) {
```

Comment:
when I have a feat. product section and change the variant, it should change the image and show the associated image I believe. Atm it's throwing an error about: `Cannot read properties of null (reading 'setActiveMedia')`

I think it's because this code was moved from global.js and the ` updateMedia()` function to this file.

Reply:
Oh no 🤦‍♂️ That's not good news. I'll need to do some refactor and add the `media-gallery` component to the product page tomorrow morning. 

Thanks for flagging this - it's a really important bug that I completely missed. Glad that you found this before we merged ❤️

Reply:
Just fixed. `media-gallery` is now a shared script between `main-product` and `featured-product`. I've added a `<media-gallery>` component to `featured-product` and made its media list a `GalleryViewer-` element, so the function will now work as expected.

I've also fixed a bug with low resolution images being served on featured product due to missing `media_width` variable.